### PR TITLE
CATROID-1118 && CATROID-1117 [GSOC21]

### PR DESCRIPTION
**Pull request by ndrnour**
[CATROID-1117](https://jira.catrob.at/browse/CATROID-1117): Refactor ProjectActivity and its related fragments to kotlin
[CATROID-1118](https://jira.catrob.at/browse/CATROID-1118):  Refactor ProjectListActivity and its related fragments to kotlin

### Your checklist for this pull request
Please review the [contributing guidelines](https://github.com/Catrobat/Catroid/blob/develop/README.md) and [wiki pages](https://github.com/Catrobat/Catroid/wiki/) of this repository.

- [ ] Include the name of the Jira ticket in the PR’s title
- [ ] Include a summary of the changes plus the relevant context
- [ ] Choose the proper base branch (*develop*)
- [ ] Confirm that the changes follow the project’s coding guidelines
- [ ] Verify that the changes generate no compiler or linter warnings
- [ ] Perform a self-review of the changes
- [ ] Verify to commit no other files than the intentionally changed ones
- [ ] Include reasonable and readable tests verifying the added or changed behavior
- [ ] Confirm that new and existing unit tests pass locally
- [ ] Check that the commits’ message style matches the [project’s guideline](https://github.com/Catrobat/Catroid/wiki/Commit-Message-Guidelines)
- [ ] Stick to the project’s gitflow workflow
- [ ] Verify that your changes do not have any conflicts with the base branch
- [ ] After the PR, verify that all CI checks have passed
- [ ] Post a message in the *catroid-stage* or *catroid-ide* [Slack channel](https://catrobat.slack.com) and ask for a code reviewer

### CODE REVIEW

**Review by Walcho1125 — CHANGES_REQUESTED** [review]
Please take a look at the remarks, try to avoid using companion objects just for class intern locally used constants like in ProjectActivity.kt etc. as the overhead for companion objects are higher than just declaring them as file-private constants.

File: catroid/src/main/java/org/catrobat/catroid/ui/ProjectActivity.kt [hunk 1]
```diff
@@ -0,0 +1,433 @@
...
+        val TAG = ProjectActivity::class.java.simpleName
+        const val FRAGMENT_SCENES = 0
+        const val FRAGMENT_SPRITES = 1
+
+        const val SPRITE_POCKET_PAINT = 0
+        const val SPRITE_LIBRARY = 1
+        const val SPRITE_FILE = 2
+        const val SPRITE_CAMERA = 3
+        const val SPRITE_OBJECT = 4
+        const val EXTRA_FRAGMENT_POSITION = "fragmentPosition"
```
[the remark this answers is not in the document]
Comment (ndrnour) [reply catroid/src/main/java/org/catrobat/catroid/ui/ProjectActivity.kt]: fixed

**Comment by ndrnour** [comment]
rebased and fixed the merge conflicts and got rid of companion objects.

**Review by Walcho1125 — APPROVED** [review]
Looks good to me!

**Pull request closed without merging**